src/main/scala/fi/oph/koski/validation/Lukio2019ArvosanaValidation.scala

```diff
@@ -18,47 +18,58 @@ object Lukio2019ArvosanaValidation {
     ))
   }
 
-  private def validateVieraatKielet(suoritus: LukionPäätasonSuoritus2019): HttpStatus = {
+  private def validateVieraatKielet(suoritus: Suoritus): HttpStatus = {
     val statii = suoritus.osasuoritukset.toList.flatten.map({
       case k:LukionOppiaineenSuoritus2019
         if vieraatKielet.contains(k.koulutusmoduuli.tunniste.koodiarvo) &&
           k.koulutusmoduuli.pakollinen &&
           k.arviointi.toList.flatten.exists(a => kirjainarvosanat.contains(a.arvosana.koodiarvo)) =>
         KoskiErrorCategory.badRequest.validation.arviointi.sallittuVainValinnaiselle(s"Pakollisen vieraan kielen oppiaineen ${suorituksenTunniste(k)} arvosanan pitää olla numero")
+      case k:LukionOppiaineenPreIBSuoritus2019
```

Comment:
Tähän ei ole tehty ylimääräistä yhteistä kanta-traittia (`LukiomaisestiValidoituvaOppiaineenSuoritus2019` tms.), koska sitten pitäisi lisätä yhteisiä kanta-traitteja jokaiselle `LukionOppiaineenSuoritus2019`n ja `LukionOppiaineenPreIBSuoritus2019`n kentälle ja vielä niidenkin sisäisille kentille jne. 

Kannattaisiko kuitenkin tehdä? Oma fiilis on tällä hetkellä se, että traitteja on vaan jo nyt paljon, ja käytännössä voi olla selkeämpää tehdä näin, että validoinneissa on sitten vähän duplikaattikoodia.

Jos haluttaisiin yhdistää, niin ehkä selkeintä olisi luopua kokonaan tavasta ryhmitellä Pre-IB:ssä(kin) käytettävät lukion luokat erillisillä `PreIB...`-traiteilla, ja käyttää lukion traitteja ja luokkia suoraan Pre-IB-koodissa. Tämä varmaan toimisi tällä hetkellä, koska lukion kaikki luokat ovat käytössä Pre-IB-puolella, eikä niitä ole karsittu.

Tämä olisi vaan sitten eri ratkaisu kuin vanhoissa Pre-IB-opinnoissa, ja saattaisi piilottaa koodin jatkoylläpitäjiltä Pre-IB-riippuvuuden nykyistäkin pahemmin.

Reply:
näitä vastaavia duplikaattipätkiä on tiedostossa `Lukio2019VieraatKieletValidation.scala` lisää.